lib/spack/spack/environment/environment.py

```diff
@@ -1247,7 +1247,10 @@ def include_concrete_envs(self):
 
     def destroy(self):
         """Remove this environment from Spack entirely."""
-        shutil.rmtree(self.path)
+        if os.path.islink(self.path):
+            os.unlink(self.path)
+        else:
+            shutil.rmtree(self.path)
```

Comment:
This logic will falter if Windows does not have symlinking enabled. Spack's fallback strategy is to utilize junctions or hardlinks, both of which will trip up the logic here (`os.path.islink` will return false and then `shutil.rmtree` will throw as hardlinks and junctions are not trees). 

I really thought we had a solution for that in the symlink module, but evidently not. IMO this is a failing of the Windows side, not this PR. I'll make a fix for that ASAP so this PR can move forward.

Reply:
On further inspection we actually do have alternatives to readlink/islink/etc in the `llnl.util.symlink` module that are better suited for cross platform use, we should prefer those here.

Reply:
Switched to using `islink` from `llnl.util.symlink` to address Windows compatibility feedback. @johnwparent let me know if this now looks good to me.